### PR TITLE
feat(TDL-46): add functionality of creating new group of tasks

### PR DESCRIPTION
**Pull request by macopul**
Added feature to add accordion with list of tasks. 
There is aborted idea of adding taks without group ( You can add task, but can not delete or update it)
Please do not look too inquisitively at styling

### CODE REVIEW

File: src/helpers/storage.js [hunk 1]
```diff
@@ -1,34 +1,135 @@
...
+  );
+  foundGroupOfTasks.tasks = newListOfTasksInFoundGroupOfTasks;
+  const updatedGroupsOfTasks = currentGroupOfTasks.map((group) =>
+    group.id === foundGroupOfTasks.id ? foundGroupOfTasks : group,
+  );
+
+  saveDataInStorageForKey(STORAGE_GROUP_OF_TASKS_KEY, updatedGroupsOfTasks);
+};
+
+// THIS IS ANOTHER SOLUTION OF deleteTaskInGroupStorage LEFT BY PURPOSE
```
Comment (woopul) [inline src/helpers/storage.js:89]: In code base we do not leave that kind of comments - please remove, before that you can create another branch which will not be merged, but you could still switch and get back to this if you will need

File: src/helpers/storage.js [hunk 2]
```diff
@@ -1,34 +1,135 @@
...
+    currentTask.id === task.id ? task : currentTask,
+  );
+  saveDataInStorageForKey(STORAGE_TASK_KEY, updatedTasks);
+};
+
+export const getGroupOfTasksDataFromLocalStorage = () => {
+  return JSON.parse(localStorage.getItem(STORAGE_GROUP_OF_TASKS_KEY)) || [];
+};
+
+// THIS IS ANOTHER SOLUTION OF updateTaskInGroupStorage LEFT BY PURPOSE
```
Comment (woopul) [inline src/helpers/storage.js:37]: In code base we do not leave that kind of comments - please remove, before that you can create another branch which will not be merged, but you could still switch and get back to this if you will need

File: src/helpers/storage.js [hunk 3]
```diff
@@ -1,34 +1,135 @@
...
+
+  saveDataInStorageForKey(STORAGE_GROUP_OF_TASKS_KEY, updatedAllGroups);
+};
+
+export const deleteTaskInGroupStorage = (task, id) => {
+  const currentGroupOfTasks = getGroupOfTasksDataFromLocalStorage();
+  const foundGroupOfTasks = [...currentGroupOfTasks].find((group) => group.id === task.groupId);
+  const indexOfDeletedTask = foundGroupOfTasks.tasks.findIndex((tasks) => tasks.id === id);
+  const newListOfTasksInFoundGroupOfTasks = foundGroupOfTasks.tasks.filter(
+    (tasksList) => tasksList !== foundGroupOfTasks.tasks[indexOfDeletedTask],
```
Comment (woopul) [inline src/helpers/storage.js:79]: You filter through tasks by group.tasks.filter - inside of which parameter is actually task, not taskList as you indicate it here - please change name for any generic item/element/taskItem or task. We should not be confused about the type of variables and should aim to name them as precisely as we can by Cleancode rules.

File: src/helpers/storage.js [hunk 4]
```diff
@@ -1,34 +1,135 @@
...
+    return group;
+  });
+
+  saveDataInStorageForKey(STORAGE_GROUP_OF_TASKS_KEY, updatedAllGroups);
+};
+
+export const deleteTaskInGroupStorage = (task, id) => {
+  const currentGroupOfTasks = getGroupOfTasksDataFromLocalStorage();
+  const foundGroupOfTasks = [...currentGroupOfTasks].find((group) => group.id === task.groupId);
+  const indexOfDeletedTask = foundGroupOfTasks.tasks.findIndex((tasks) => tasks.id === id);
```
Comment (woopul) [inline src/helpers/storage.js:77]: What youre doing here is finding index of task inside array to delete, but you already have id of task to delete. Only thing you need check is in line 79 - if ID of currentlyFilteredElement !== ID of element to delete. By doing what you did you have one extra layer of logic (for loops/maps/filters) on big sets of data are expensive operations what influence performance. You searched for information you actually do not need.

**Review by woopul — CHANGES_REQUESTED** [review]
Overall logic and cleanance looks good, good work. Some small issues to fix

File: src/helpers/storage.js (hunk 1 above)
Comment (macopul) [reply src/helpers/storage.js]: done

File: src/helpers/storage.js (hunk 2 above)
Comment (macopul) [reply src/helpers/storage.js]: done

File: src/helpers/storage.js (hunk 3 above)
Comment (macopul) [reply src/helpers/storage.js]: done

File: src/helpers/storage.js (hunk 4 above)
Comment (macopul) [reply src/helpers/storage.js]: done

**Review by woopul — APPROVED** [review]

**Pull request merged**